doc/user/metrics/README.md

```diff
@@ -13,44 +13,120 @@ By default, the metrics are served on `0.0.0.0:8383/metrics`. To modify the port
-        mgr, err := manager.New(cfg, manager.Options{
-            Namespace:          namespace,
-            MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
-        })
+       // Pass metrics address to controller-runtime manager
+       // Create a new Cmd to provide shared dependencies and start components
+       mgr, err := manager.New(cfg, manager.Options{
+          Namespace: namespace,
+          MapperProvider:     restmapper.NewDynamicRESTMapper,
+          MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+       })
 
         ...
 
         // Add to the below struct any other metrics ports you want to expose.
-	    servicePorts := []v1.ServicePort{
-		    {Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
-	    }
+      	servicePorts := []v1.ServicePort{
+      		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+      		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
+      	}
+      	// Create Service object to expose the metrics port(s).
+      	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+      	if err != nil {
+      		log.Info("Could not create metrics Service", "error", err.Error())
+      	}
+      
+      	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
+      	// necessary to configure Prometheus to scrape metrics from this operator.
+      	services := []*v1.Service{service}
+      	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+      	if err != nil {
+      		log.Info("Could not create ServiceMonitor object", "error", err.Error())
+      		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
+      		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
+      		if err == metrics.ErrServiceMonitorNotPresent {
+      			log.Info("Install prometheus-operator in your cluster to create ServiceMonitor objects", "error", err.Error())
+      		}
+      	}
+      
+      	log.Info("Starting the Cmd.")
+      
+      	// Start the Cmd
+      	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
+      		log.Error(err, "Manager exited non-zero")
+      		os.Exit(1)
+      	}
 
-        // Create Service object to expose the metrics port.
-        _, err = metrics.CreateMetricsService(context.TODO(), cfg, servicePorts)
+    }
+    
+    // serveCRMetrics gets the Operator/CustomResource GVKs and generates metrics based on those types.
+    // It serves those metrics on "http://metricsHost:operatorMetricsPort".
+    func serveCRMetrics(cfg *rest.Config) error {
```

Comment:
This function is not relevant to the `General metrics` section that this usage block is in.

We should add a code snippet in the `Custom resource specific metrics` section further down.